src/setup-gcloud.ts

```diff
@@ -100,17 +100,33 @@ export async function run(): Promise<void> {
         credsPath = path.join(githubWorkspace, uniqueName);
       }
 
-      const serviceAccountKeyObj = parseServiceAccountKey(serviceAccountKey);
-      await writeSecureFile(
-        credsPath,
-        JSON.stringify(serviceAccountKeyObj, null, 2), // Print to file as string w/ indents
-      );
+      // If explicit SA input, parse it, write to disk and set GCLOUD_PROJECT.
+      if (serviceAccountKey) {
+        const serviceAccountKeyObj = parseServiceAccountKey(serviceAccountKey);
+        await writeSecureFile(
+          credsPath,
+          JSON.stringify(serviceAccountKeyObj, null, 2), // Print to file as string w/ indents
+        );
+        core.exportVariable('GCLOUD_PROJECT', serviceAccountKeyObj.project_id);
+      } else if (process.env.GOOGLE_GHA_CREDS_PATH) {
+        // No explicit SA input but process.env.GOOGLE_GHA_CREDS_PATH is set.
+        // process.env.GOOGLE_GHA_CREDS_PATH already contains credentials written to disk,
+        // so set credsPath to the existing cred filepath.
+        credsPath = process.env.GOOGLE_GHA_CREDS_PATH;
+        // User is likely using google-github-actions/auth for auth followed by setup-gcloud with export_default_credentials.
+        // This is unnecessary as auth already exports credentials.
+        core.warning(
```

Comment:
`setup-gcloud` also exports `GOOGLE_GHA_CREDS_PATH` (maybe we should remove that?), so I think you'd hit this block with back-to-back `export_default_credentials` for two setup-gcloud uses.

We should add an integration test to make sure back-to-back setup-gcloud works and uses the correct auth:

```yaml
- uses: setup-gcloud
  with:
    credentials: cred1

- run: gcloud auth list | gregp 'cred1email'

- uses: setup-gcloud
  with:
    credentials: cred2

- run: gcloud auth list | gregp 'cred2email'
```

Reply:
I _think_ it'll be fine because it'll fall into the first case of this conditional, but it gets weird if the user does:

```yaml
- uses: setup-gcloud
  with:
    credentials: cred1

- run: gcloud auth list | gregp 'cred1email'

# I think this will throw the warning
- uses: setup-gcloud
```

Reply:
Yeah the first case would be fine 

For
```yaml
- uses: setup-gcloud
  with:
    credentials: cred1
    
- uses: setup-gcloud
``` 
since user is not specifying explicit creds, it is no op.
Using something else other than `GOOGLE_GHA_CREDS_PATH` for cleanup in setup-gcloud is also feasible.